components/claim/components/email_to_ldap.tsx

```diff
@@ -0,0 +1,259 @@
+                if (data.follow_link) {
+                    window.location.href = data.follow_link;
+                }
+            },
+            (err: {server_error_id: string; id: string; message: string}) => {
+                if (!showMfa && err.server_error_id === ClaimErrors.MFA_VALIDATE_TOKEN_AUTHENTICATE) {
+                    setShowMfa(true);
+                } else {
+                    switch (err.id) {
+                    case ClaimErrors.ENT_LDAP_LOGIN_USER_NOT_REGISTERED:
+                    case ClaimErrors.ENT_LDAP_LOGIN_USER_FILTERED:
+                    case ClaimErrors.ENT_LDAP_LOGIN_MATCHED_TOO_MANY_USERS:
+                        setLdapError(err.message);
+                        setShowMfa(false);
+                        break;
+                    case ClaimErrors.ENT_LDAP_LOGIN_INVALID_PASSWORD:
+                        setLdapPasswordError(err.message);
+                        setShowMfa(false);
+                        break;
+                    case ClaimErrors.API_USER_INVALID_PASSWORD:
+                        setPasswordError(err.message);
+                        setShowMfa(false);
+                        break;
+                    default:
+                        setServerError(err.message);
+                        setShowMfa(false);
+                    }
+                }
+            },
+        );
+    };
+    let serverErrorElement: JSX.Element | null = null;
+    let formClass = 'form-group';
+    if (serverError) {
+        serverErrorElement = <div className='form-group has-error'><label className='control-label'>{serverError}</label></div>;
+        formClass += ' has-error';
+    }
+
+    let passwordErrorElement: JSX.Element | null = null;
+    let passwordClass = 'form-group';
+    if (passwordError) {
+        passwordErrorElement = <div className='form-group has-error'><label className='control-label'>{passwordError}</label></div>;
+        passwordClass += ' has-error';
+    }
+
+    let ldapErrorElement: JSX.Element | null = null;
+    let ldapClass = 'form-group';
+    if (ldapError) {
+        ldapErrorElement = <div className='form-group has-error'><label className='control-label'>{ldapError}</label></div>;
+        ldapClass += ' has-error';
+    }
+
+    let ldapPasswordErrorElement: JSX.Element | null = null;
+    let ldapPasswordClass = 'form-group';
+    if (ldapPasswordError) {
+        ldapPasswordErrorElement = <div className='form-group has-error'><label className='control-label'>{ldapPasswordError}</label></div>;
+        ldapPasswordClass += ' has-error';
+    }
+
+    const loginPlaceholder = ldapLoginFieldName || localizeMessage('claim.email_to_ldap.ldapId', 'AD/LDAP ID');
+
+    if (showMfa) {
+        return (
+            <LoginMfa
+                loginId={email}
+                password={password}
+                title={{id: t('claim.email_to_ldap.title'), defaultMessage: 'Switch Email/Password Account to AD/LDAP'}}
```

Comment:
could we please not paste in object literals into the props? It is better to define it beforehand and pass in the variable. The same applies to some of the other components in the second `return`

Reply:
Done

Reply:
Changed all of them